### ISSUE TITLE
Enhancement: add an option `logging_module` to `LoggingConfig`

### ISSUE THREAD

**Issue by jderrien** [litestar-org/litestar#3536]
### Summary

It would permit to switch from `picologging` to `logging` without having to hack around or uninstall `picologging`.
It would be useful in tests but also in production (vs having to update a requirements file and rebuild an artifact).

This option would be optional. The default behavior would be the same as today (use `picologging` if the lib is installed).

### Basic Example

```
std_logging_config = LoggingConfig(logging_module="logging")
picologging_logging_config = LoggingConfig(logging_module="picologging")
```

**Comment by provinzkraut** [comment]
Sounds good to me. Would you be interested in providing a PR for this?

**Comment by jderrien** [comment]
@provinzkraut Yes! I already have a branch for this work. It's based on #3543 so we just need to merge this one before a send a PR. 👍

**Comment by provinzkraut** [comment]
> @provinzkraut Yes! I already have a branch for this work. It's based on #3543 so we just need to merge this one before a send a PR. 👍

Nice! Merged that one :)
